### PR TITLE
Added outputs to the reporters. Moved *each to ut_test

### PR DESCRIPTION
**Pull request by Pazus**
Now `ut_executable` stores the stderr, stdout of the execution wich can be used by reporters.
`ut_documentations_reporter` prints the output of its steps along with test results (in future we might disable that by parameter)
`ut_teamcity_reporter` also reports output and stderr according to the Teamcity messaging format.
Added new tests to check new behavior

Moved beforeeach/aftereach executables from `ut_suite` to `ut_test`, fixed the behavior on failure.

### CODE REVIEW

**Comment by jgebal** [comment]
@pazus, 
Here is an idea on how to handle before/after calls as a list.

```sql
l_before_each_list := ut_executables(
  ut_executable('the before each'),
  ut_executable(null) -- no before test defined
);
l_after_each_list := ut_executables(
  ut_executable('the after test'),
  ut_executable(null) -- no after test
);

--execute the before blocks
i := l_before_each_list.first;
while i is not null loop
  if l_before_each_list(i) is not null then 
    l_before_each_list(i).do_execute;
  end if;
  l_before_exec_count := l_before_exec_count + 1;
  i := l_before_each_list.next(i);
end loop;

-- then you track the count of executed before block and execute
-- the after blocks in following loop
i := l_after_each_list.last;
while i is not null loop
  if l_after_each_list(i) is not null then 
    l_after_each_list(i).do_execute;
  end if;
  l_after_exec_count := l_after_exec_count + 1;
  exit when l_after_exec_count >= l_before_exec_count;
  i := l_after_exec_count .prior(i);
end loop;
```

That is a rough idea - not a real implementation, but this would also open doors for propagating `beforeeach` from parents (parent suites or within suite when we introduce sub-suites/contexts).

What do you think about this?

**Comment by Pazus** [comment]
Should I implement it in the current PR?

File: source/reporters/ut_documentation_reporter.tpb [hunk 1]
```diff
@@ -44,6 +44,17 @@ create or replace type body ut_documentation_reporter is
 
   overriding member procedure after_calling_test(self in out nocopy ut_documentation_reporter, a_test ut_test) as
     l_message varchar2(4000);
+    
+    procedure print_output(a_exectable ut_executable) is
+      l_lines ut_varchar2_list;
+    begin
+      if a_exectable is not null and a_exectable.is_defined and a_exectable.serveroutput is not null and dbms_lob.getlength(a_exectable.serveroutput) > 0 then
```
Comment (jgebal) [inline source/reporters/ut_documentation_reporter.tpb]: typo in var name

File: source/core/ut_suite_manager.pkb [hunk 2]
```diff
@@ -174,7 +172,9 @@ create or replace package body ut_suite_manager is
                 a_rollback_type => l_rollback_type,
                 a_ignore_flag   => l_proc_annotations.exists('disabled'),
                 a_before_test_proc_name => l_setup_procedure,
-                a_after_test_proc_name  => l_teardown_procedure
+                a_after_test_proc_name  => l_teardown_procedure,
```
Comment (jgebal) [inline source/core/ut_suite_manager.pkb]: can you cleanup the variable names to match new conventions?

File: source/reporters/ut_documentation_reporter.tpb [hunk 3]
```diff
@@ -44,6 +44,17 @@ create or replace type body ut_documentation_reporter is
 
   overriding member procedure after_calling_test(self in out nocopy ut_documentation_reporter, a_test ut_test) as
     l_message varchar2(4000);
+    
+    procedure print_output(a_exectable ut_executable) is
```
Comment (jgebal) [inline source/reporters/ut_documentation_reporter.tpb]: Can we handle printing outputs of `beforeall` `afterall` as well?

File: tests/ut_test/ut_test.TestOutputGathering.sql [hunk 4]
```diff
@@ -0,0 +1,88 @@
+create or replace package ut_output_tests
+as
+ --%suite
+
```
Comment (jgebal) [inline tests/ut_test/ut_test.TestOutputGathering.sql]: Can we add beforeall/afterall?

**Comment by Pazus** [comment]
@jgebal Thanks for the comments, I've addressed them.

**Pull request merged**